java/src/jmri/util/swing/GuiUtilBase.java

```diff
@@ -34,14 +34,21 @@ static Action actionFromNode(Element child, WindowInterface wi, Object context)
             return createEmptyMenuItem(null, "<none>");
         }
         name = LocaleSelector.getAttribute(child, "name");
-        if ((name == null) || (name.equals(""))) {
+        if ((name == null) || (name.isEmpty())) {
             if (child.getChild("name") != null) {
                 name = child.getChild("name").getText();
             }
         }
 
-        if (child.getChild("icon") != null) {
-            icon = new ImageIcon(FileUtil.findURL(child.getChild("icon").getText()));
+        Element childUrl = child.getChild("icon"); 
+        if ( childUrl != null) { // NOI18N
+            java.net.URL iconUrl =  FileUtil.findURL(childUrl.getText()); // NOI18N
```

Comment:
No need for `// NOI18N` in either of these lines. `// NOI18N` is used for constant strings, like `"MyTurnout"`.

Except that, everything looks fine.

Reply:
Thanks @danielb987 , good catch, should have been on line above, have uploaded update.